six/modules/c++/six/source/XMLControl.cpp

```diff
@@ -44,14 +44,16 @@ void XMLControl::loadSchemaPaths(std::vector<std::string>& schemaPaths)
 {
     if (schemaPaths.empty())
     {
+        const sys::OS os;
+
         std::string envPath;
-        sys::OS().getEnvIfSet(six::SCHEMA_PATH, envPath);
+        os.getEnvIfSet(six::SCHEMA_PATH, envPath);
         str::trim(envPath);
         if (!envPath.empty())
         {
             schemaPaths.push_back(envPath);
         }
-        else
+        else if (os.exists(DEFAULT_SCHEMA_PATH))
         {
             schemaPaths.push_back(DEFAULT_SCHEMA_PATH);
         }
```

Comment:
I'd log a warning in validate() when loadSchemaPaths() returns no results and the user didn't provide any of their own. I'm OK that loadSchemaPaths() doesn't try to load non-existing paths.

Reply:
Also, it's not portable, but for *nix we could set this up to readlink('/proc/self/exe') to find the location of the current executable, then try a few locations relative to that instead of using a build-time DEFAULT_SCHEMA_PATH. There's probably a windows way to do that too. 

I'm satisfied with what's here (plus the log warning) for the purposes if getting things fixed, though.

Reply:
Updated to log a warning message - also guarding against a null logger.